src/Microsoft.Diagnostics.Monitoring.RestServer/MetricsOptions.cs

```diff
@@ -47,7 +47,7 @@ public MetricsOptions()
         
         public string Endpoints { get; set; }
 
-        public int?[] Ports => _ports.Value;
+        public int?[] Ports => Enabled ? _ports.Value : Array.Empty<int?>();
```

Comment:
Note this change is to fix a minor issue where there is no metrics but Configuration.Bind attempts to read the Ports property.

Reply:
In `DiagnosticsMonitorCommandHandler.CreateWebHostBuilder`, I see that it is adding metrics to the configuration conditionally. Could you instead always add metrics but set the Enabled configuration option to correspond to the metrics parameter? Then this fix might not be needed.

Reply:
I am making some more configuration changes in my next PR, so I will actually undo this change for now and push your suggestion to that PR.